### PR TITLE
Add Material Design

### PR DESCRIPTION
**Pull request by juancarlospaco**
nimwc-themes

**Plugin Store with the Darkest theme of all:**

nimwc-darkest

- Includes Material Design.
- Includes Dark Theme.
- Includes Solarized Theme.
- I tried all themes one-by-one.
- Previews are disabled iframes with static HTML so they will render exactly as preview across browsers.
- Wont break as much as I expected, but from now on we should stop using inline style colors and [use bulma classes](https://bulma.io/documentation/modifiers/typography-helpers/), you can use all other styles inline, just not hardcoded colors.

:cool:

### CODE REVIEW

File: nimwcpkg/tmpl/main.tmpl [hunk 1]
```diff
@@ -21,7 +21,7 @@
   <link rel="icon" type="image/png" href="/images/logo/favicon-32x32.png" sizes="32x32">
   <link rel="icon" type="image/png" href="/images/logo/favicon-192x192.png" sizes="192x192">
   <link rel="apple-touch-icon" sizes="180x180" href="/images/logo/favicon-180x180.png">
-  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bulma/0.7.4/css/bulma.min.css">
+  ${ bulma(c) }
```
Comment (ThomasTJdev) [inline nimwcpkg/tmpl/main.tmpl]: tmpl/main.tmpl(24, 15) Error: undeclared identifier: 'bulma'

Comment (juancarlospaco) [reply nimwcpkg/tmpl/main.tmpl]: `-d:rc` :grey_question: 
And wipe clean Database too.

**Comment by juancarlospaco** [comment]
The thing with this being a plugin is that it will load twice the CSS needed.

Because this replaces the default CSS for Bulma, with a Themed one, then if its a plugin, it will load the default from the standarddata header, then will load a themed one from the plugin.

We kinda kill the page size performance there.
But themes are per-user, then everyone starts with vanilla theme, and changes it for that user only, if the user wants to, and Admin can Reset it too.

This works everywhere, not only on Admin page.
:thinking:

File: nimwcpkg/tmpl/main.tmpl (hunk 1 above)
Comment (ThomasTJdev) [reply nimwcpkg/tmpl/main.tmpl]: Does not work with `-d:rc`. What do you mean with wipe DB?

```
#ALTER TABLE person ADD COLUMN theme VARCHAR(300);

[/nim_websitecreator] - [material] $ nim c -r -d:dev -d:ssl -d:rc nimwc

Plugins - required proc:
proc extensionUpdateDB*(db: DbConn) =
  discard
tmpl/main.tmpl(24, 15) Error: undeclared identifier: 'bulma'
```

**Comment by juancarlospaco** [comment]
This will reborn someday, maybe as a Plugin, Closing.

**Pull request closed without merging**